tests/scripts/kubeadm.sh

```diff
@@ -30,12 +29,48 @@ install() {
     done
 
     if [ "${kube_ready}" == "False" ]; then
-        echo "k8s node never went to Ready status"
+        echo "k8s master node never went to Ready status"
         exit 1
     fi
 
-    echo "k8s node in Ready status"
+    echo "k8s master node in Ready status"
+}
+install_node(){
+    echo "inside install node function"
+    echo "kubeadm join ${1} ${2} ${3} --skip-preflight-checks"
+    sudo kubeadm join ${1} ${2} ${3} --skip-preflight-checks || true
+}
+wait_for_ready(){
+    #expect 3 node cluster by default
+    local numberOfNode=${1:-3}
+    local count=0
+    sudo cp /etc/kubernetes/admin.conf $HOME/
+    sudo chown $(id -u):$(id -g) $HOME/admin.conf
+    export KUBECONFIG=$HOME/admin.conf
+
+    until [[ $count -eq $numberOfNode ]]; do
+        echo "wait for K8s node $count to be Ready"
+        kube_ready=$(kubectl get node -o jsonpath='{.items['$count'].status.conditions[3].status}')
+        INC=0
+        until [[ "${kube_ready}" == "True" || $INC -gt 30 ]]; do
+            echo "."
```

Comment:
As coded this echo isn't doing anything useful other than adding an extra line with a single '.' on it.  A few lines down it echoes the status after every check, also on a separate line.  Also, 30 seconds between checks seems like a long time.  My suggestions would be:
1) Change this line to:
    echo -n "."
so that all the "."s are on one line.
2) Get rid of the echo at line 60 so there are only the "."s
3) Change the loop from 30 loops of 30 seconds/loop to 90 loops of 10 seconds per loop
4) And add a final "echo" after the loop to terminate the list of "."s.